src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -414,49 +416,104 @@ case class JesBackend(actorSystem: ActorSystem)
   }
 
   /**
-   * Creates a set of JES inputs for a backend call.
-   * Note that duplicates input files (same gcs path) will be (re-)localized every time they are referenced.
-   */
+    * Turns WdlFiles into relative paths.  These paths are relative to the working disk
+    *
+    * relativeLocalizationPath("foo/bar.txt") -> "foo/bar.txt"
+    * relativeLocalizationPath("gs://some/bucket/foo.txt") -> "some/bucket/foo.txt"
+    */
+  private def relativeLocalizationPath(file: WdlFile): WdlFile = {
+    GcsPath.parse(file.value) match {
+      case Success(gcsPath) => WdlFile(gcsPath.bucket + "/" + gcsPath.objectName, file.isGlob)
+      case Failure(e) => file
+    }
+  }
+
   def generateJesInputs(backendCall: BackendCall): Iterable[JesInput] = {
-    backendCall.locallyQualifiedInputs mapValues { _.collectAsSeq { case w: WdlFile => w } } flatMap {
-      case (name, files) => jesInputsFromWdlFiles(name, files, files map { gcsPathToLocal(_).asInstanceOf[WdlFile] })
+    /**
+      * Commands in WDL tasks can also generate input files.  For example: ./my_exec --file=${write_lines(arr)}
+      *
+      * write_lines(arr) would produce a string-ified version of the array stored as a GCS path.  The next block of code
+      * will use the file evaluator on each expression in the command to write these files and return back paths to
+      * those files so JES can localize them
+      */
+    val commandExpressions = backendCall.call.task.commandTemplate.collect({
+      case x: ParameterCommandPart => x.expression
+    })
+
+    val commandFiles = commandExpressions.map({ e =>
+      val k = e.toWdlString.md5Sum.substring(0, 8)
```

Comment:
`md5Sum.substring(0, 8)` is used in a couple of spots, is there one name corresponding to this concept that would allow it to be turned into a function?